src/tests/tracing/eventpipe/processinfo/processinfo.cs

```diff
@@ -0,0 +1,201 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Diagnostics.Tools.RuntimeClient;
+using Microsoft.Diagnostics.Tracing;
+using Tracing.Tests.Common;
+
+namespace Tracing.Tests.ProcessInfoValidation
+{
+    public class ProcessInfoValidation
+    {
+        public static string NormalizeCommandLine(string cmdline)
+        {
+            // ASSUMPTION: double quotes (") and single quotes (') are used for paths with spaces
+            // ASSUMPTION: This test will only have two parts to the commandline
+
+            // check for quotes in first part
+            var parts = new List<string>();
+            bool isQuoted = false;
+            int start = 0;
+
+            for (int i = 0; i < cmdline.Length; i++)
```

Comment:
If we are to consider every single possibly case, it's possible to have both single and double quotes are part of file and directory names. So you'd have to check that the single or double quote is not escaped e.g. \\' or \\". But then you also have to check that the escaping character is not escaped either e.g. \\\\. While it's not a complete recursive mess, it will be tricky to implement.